locking/cache.go

```diff
@@ -0,0 +1,101 @@
+package locking
+
+import (
+	"strings"
+
+	"github.com/git-lfs/git-lfs/tools/kv"
+)
+
+const (
+	// We want to use a single cache file for integrity, but to make it easy to
+	// list all locks, prefix the id->path map in a way we can identify (something
+	// that won't be in a path)
+	idKeyPrefix string = "*id*://"
+)
+
+type LockCache struct {
+	kv *kv.Store
+}
+
+func NewLockCache(filepath string) (*LockCache, error) {
+	kv, err := kv.NewStore(filepath)
+	if err != nil {
+		return nil, err
+	}
+	return &LockCache{kv}, nil
+}
+
+func (c *LockCache) encodeIdKey(id string) string {
+	// Safety against accidents
+	if !c.isIdKey(id) {
+		return idKeyPrefix + id
+	}
+	return id
+}
+func (c *LockCache) decodeIdKey(key string) string {
+	// Safety against accidents
+	if c.isIdKey(key) {
+		return key[len(idKeyPrefix):]
+	}
+	return key
+}
+func (c *LockCache) isIdKey(key string) bool {
+	return strings.HasPrefix(key, idKeyPrefix)
+}
+
+// Cache a successful lock for faster local lookup later
+func (c *LockCache) CacheLock(l Lock) error {
```

Comment:
I think having functions with a `Cache` prefix is redundant. Especially in the `*Client` usage: `c.cache.CacheUnlockById(id)`. What do you think about method names like this:

* `CacheLock` -> `Set`
* `CacheUnlockByPath` -> `DeleteByPath`
* `CacheUnlockById` -> `DeleteById`
* `CachedLocks` -> `GetAll` or just `All`